src/robotFactory.js

```diff
@@ -1,10 +1,82 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    if (Object.keys(coords).length === 0) {
+      this.coords = {
+        x: 0, y: 0,
+      };
+    } else {
+      this.coords = { ...coords };
+    }
+  }
```

Comment:
You can greatly simplify this action:
```suggestion
    this.coords = {
      x: coords.x || 0,
      y: coords.y || 0,
    };
```